rmw_fastrtps_shared_cpp/src/rmw_subscription.cpp

```diff
@@ -106,4 +110,155 @@ __rmw_subscription_get_actual_qos(
 
   return RMW_RET_OK;
 }
+
+rmw_ret_t
+__rmw_subscription_set_content_filter(
+  rmw_subscription_t * subscription,
+  const rmw_subscription_content_filter_options_t * options
+)
+{
+  auto info = static_cast<CustomSubscriberInfo *>(subscription->data);
+  eprosima::fastdds::dds::ContentFilteredTopic * filtered_topic = info->filtered_topic_;
+  const bool filter_expression_empty = (*options->filter_expression == '\0');
+
+  if (!filtered_topic && filter_expression_empty) {
+    // can't reset current subscriber
+    RMW_SET_ERROR_MSG(
+      "current subscriber has no content filter topic");
```

Comment:
> so instead of getting a runtime error you get a compile time error.

What do you mean?

Reply:
If the prototype would be:

```c
rmw_subscription_set_content_filter(
  rmw_cft_subscription_t * subscription,
  const rmw_subscription_content_filter_options_t * options
)
```

(note `rmw_cft_subscription_t` instead of `rmw_subscription_t`)
you wouldn't have to check this at runtime, it would be a compile time error.

i.e. related to the discussion [here](https://github.com/ros2/rmw/pull/302#discussion_r823791348)